File: 2015/Day1/Day1.py
floor = 0
position = 0
basementFound = False

inputFile = open("input.txt")
inputString = inputFile.read()

for i in range(len(inputString)):
	if inputString[i] == '(':
		floor += 1
	elif inputString[i] == ')':
		floor -= 1

	position += 1

	if floor < 0 and not basementFound:
		print("position =", position)
		basementFound = True

print("floor =", floor)

File: 2015/Day01/Day1.cpp
// 2015 Day 1
 
#include <iostream>
#include <fstream>
#include <string>

using namespace std;
 
int main()
{
  int floor = 0;
  int position = 0;
  bool basementFound = false;
  string line;

  ifstream inputFile ("input.txt");

  while (getline(inputFile, line)){

  	for (auto it = line.cbegin(); it != line.cend(); ++it){

  		if (*it == '('){
  			floor++;
  		}
  		else if (*it == ')'){
  			floor--;
  		}

      position++;

      if(floor < 0 & !basementFound){

        cout << "Floor found at position " << position << endl;
        basementFound = true;

      }
  	}
  }

  inputFile.close();

  cout << "Floor #  = " << floor << endl;

  return 0;
}